src/core/src/CoreMain.py

```diff
@@ -85,12 +89,17 @@ def __init__(self, argv):
                 telemetry_writer.write_event("EXCEPTION: " + repr(error), Constants.TelemetryEventLevel.Error)
             if status_handler is not None:
                 composite_logger.log_debug(' - Status handler pending writes flags [I=' + str(patch_installation_successful) + ', A=' + str(patch_assessment_successful) + ']')
-                if patch_operation_requested == Constants.INSTALLATION.lower() and not patch_installation_successful:
-                    status_handler.set_installation_substatus_json(status=Constants.STATUS_ERROR)
-                    composite_logger.log_debug('  -- Persisted failed installation substatus.')
-                if not patch_assessment_successful:
-                    status_handler.set_assessment_substatus_json(status=Constants.STATUS_ERROR)
-                    composite_logger.log_debug('  -- Persisted failed assessment substatus.')
+
+                if patch_operation_requested == Constants.CONFIGURE_PATCHING.lower():
+                    status_handler.set_configure_patching_substatus_json(status=Constants.STATUS_ERROR)
+                    composite_logger.log_debug('  -- Persisted failed configure patching substatus.')
+                else:
+                    if patch_operation_requested == Constants.INSTALLATION.lower() and not patch_installation_successful:
+                        status_handler.set_installation_substatus_json(status=Constants.STATUS_ERROR)
+                        composite_logger.log_debug('  -- Persisted failed installation substatus.')
+                    if not patch_assessment_successful:
+                        status_handler.set_assessment_substatus_json(status=Constants.STATUS_ERROR)
+                        composite_logger.log_debug('  -- Persisted failed assessment substatus.')
```

Comment:
This works okay if it's not configure patching that failed and threw an exception in a non-configurepatching operation.

Reply:
Updated the error logging instructions, there is a todo item marked on in line 135 of CoreMain in the latest iteration. Need to follow up on that